apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts

```diff
@@ -285,3 +307,72 @@ const parseComplianceReport = (
     url: complianceReport.url,
     description: complianceReport.description,
   };
+
+const parseQuickCheckDiscussions = (
+  quickCheckDetails: QuickCheckDetailsObject,
+) => ({
+  acknowledgedGrantNumberDetails:
+    quickCheckDetails.acknowledgedGrantNumberDetails
+      ? getLinkEntity(quickCheckDetails.acknowledgedGrantNumberDetails)
+      : null,
+  asapAffiliationIncludedDetails:
+    quickCheckDetails.asapAffiliationIncludedDetails
+      ? getLinkEntity(quickCheckDetails.asapAffiliationIncludedDetails)
+      : null,
+  availabilityStatementDetails: quickCheckDetails.availabilityStatementDetails
+    ? getLinkEntity(quickCheckDetails.availabilityStatementDetails)
+    : null,
+  codeDepositedDetails: quickCheckDetails.codeDepositedDetails
+    ? getLinkEntity(quickCheckDetails.codeDepositedDetails)
+    : null,
+  datasetsDepositedDetails: quickCheckDetails.datasetsDepositedDetails
+    ? getLinkEntity(quickCheckDetails.datasetsDepositedDetails)
+    : null,
+  labMaterialsRegisteredDetails: quickCheckDetails.labMaterialsRegisteredDetails
+    ? getLinkEntity(quickCheckDetails.labMaterialsRegisteredDetails)
+    : null,
+  manuscriptLicenseDetails: quickCheckDetails.manuscriptLicenseDetails
+    ? getLinkEntity(quickCheckDetails.manuscriptLicenseDetails)
+    : null,
+  protocolsDepositedDetails: quickCheckDetails.protocolsDepositedDetails
+    ? getLinkEntity(quickCheckDetails.protocolsDepositedDetails)
+    : null,
+});
+
+const createQuickCheckDiscussions = async (
+  environment: Environment,
+  quickCheckDetails: QuickCheckDetailsObject,
+  userId: string,
+) => {
+  const generatedDiscussions = {} as QuickCheckDetailsObject;
```

Comment:
I wonder if we could remove `parseQuickCheckDiscussions` doing some modifications like this one:

```suggestion
  const generatedDiscussions = {} as Record<
    QuickCheckDetails,
    Link<'Entry'> | null
  >;
```

and the one in the comment below, what do you think?

Reply:
Yea, it simplifies the steps. Thanks! 👌🏾